### PR TITLE
Implement settings

### PR DESCRIPTION
**Pull request by Plakhotenko**

### CODE REVIEW

File: src/AppRoutes.tsx [hunk 1]
```diff
@@ -42,6 +43,14 @@ const AppRoutes: FC = () => (
         </ProtectedRoute>
       }
     />
+    <Route
+      path="/settings"
```
Comment (Rammfall) [inline src/AppRoutes.tsx]: I guess be better to create a constant with all route names

File: src/providers/NotificationProvider/NotificationProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React, { FC, createContext, useState, useMemo } from 'react';
+
+export const NotificationContext = createContext<{
+  notificationText: string;
+  showNotification: (t: string) => void;
+}>({} as any);
```
Comment (Rammfall) [inline src/providers/NotificationProvider/NotificationProvider.tsx]: maybe create "initialContextValue" with defaults for it

File: src/pages/UserSettings/UserSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React, { FC } from 'react';
+import { Box, Container } from '@mui/material';
+import UpdatePasswordForm from '../../components/UpdatePasswordForm';
+import UpdateEmailForm from '../../components/UpdateEmailForm';
+
+const UserSettings: FC = () => (
+  <Container maxWidth="sm">
+    <Box sx={{ display: 'flex', flexDirection: 'column', gap: '32px' }}>
```
Comment (Rammfall) [inline src/pages/UserSettings/UserSettings.tsx]: Maybe use the Stack component here

Comment (Rammfall) [reply src/pages/UserSettings/UserSettings.tsx]: It has spacing prop with spacing. 

https://mui.com/material-ui/react-stack/

File: src/components/UpdatePasswordForm/UpdatePasswordForm.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    .required(),
+});
+
+const useUpdatePasswordForm = () => {
+  const [loading, setLoading] = useState(false);
+  const { showNotification } = useNotification();
+  const handleSubmit = (
+    value: IFormValue,
+    { setFieldError, resetForm }: FormikHelpers<IFormValue>
+  ) => {
```
Comment (Rammfall) [inline src/components/UpdatePasswordForm/UpdatePasswordForm.tsx:32]: What do you think, maybe create a hook with the name for example `useFormikSubmit` or similar, I don't know

**Review by Rammfall — CHANGES_REQUESTED** [review]

File: src/components/UpdatePasswordForm/UpdatePasswordForm.tsx (hunk 4 above)
Comment (Plakhotenko) [reply src/components/UpdatePasswordForm/UpdatePasswordForm.tsx:32]: good idea. but most cases has more differences than similarities.

**Pull request merged**